tests/parser/LgrOutputTests.cpp

```diff
@@ -472,3 +472,73 @@ SCHEDULE
     // Save EclipseGrid.
     eclipse_grid_file.save("OPMCARFIN-NESTED.EGRID",false,vecNNC,units);
   }
+
+  BOOST_AUTO_TEST_CASE(Test_lgr_host_cells_logical) { 
+    const std::string deck_string = R"(
+RUNSPEC
+
+DIMENS
+  3 1 1 /
+
+GRID
+
+CARFIN
+-- NAME I1-I2 J1-J2 K1-K2 NX NY NZ
+LGR1  1  3  1  1  1  1  12  1   1 1*  GLOBAL/
+ENDFIN
+
+
+DX 
+  3*1000 /
+DY
+	3*1000 /
+DZ
+	3*20 /
+
+TOPS
+	3*8325 /
+
+PORO
+  3*0.15 /
+
+PERMX
+  3*1 /
+
+COPY
+  PERMX PERMZ /
+  PERMX PERMY /
+/
+
+EDIT
+
+OIL
+GAS
+
+TITLE
+The title
+
+START
+16 JUN 1988 /
+
+PROPS
+
+REGIONS
+
+SOLUTION
+
+SCHEDULE
+)";
+    
+    Opm::Parser parser;
+    Opm::Deck deck = parser.parseString(deck_string);
+    Opm::EclipseState state(deck);
+    Opm::EclipseGrid eclipse_grid = state.getInputGrid();
+    Opm::LgrCollection lgrs = state.getLgrs();
+    // Intialize host_cell numbering.
+    eclipse_grid.init_children_host_cells();
+    std::vector<int> host_vec_sol = { 1, 1, 1, 1, 2, 2, 2, 2, 3, 3, 3, 3};
+    std::vector<int> host_vec = eclipse_grid.getLGRCell(0).get_hostnum();
+    for (std::size_t index = 0; index < host_vec.size(); index++) {
+      BOOST_CHECK_EQUAL( host_vec[index] , host_vec_sol[index]);
+    }   
```

Comment:
For what it's worth, Boost.Test has the `BOOST_CHECK_EQUAL_COLLECTIONS` macro which runs this loop for you:
```C++
BOOST_CHECK_EQUAL_COLLECTIONS(host_vec.begin(), host_vec.end(), host_vec_sol.begin(), host_vec_sol.end());
```
Using that you'd get the index of any differing elements reported in the test failure.